admin/class-plugin-availability.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * @package WPSEO\Plugin_Availability
+ */
+
+/**
+ * Class WPSEO_Plugin_Availability
+ */
+class WPSEO_Plugin_Availability {
+
+	/**
+	 * @var array
+	 */
+	protected $plugins = array();
+
+	/**
+	 * WPSEO_Plugin_Availability constructor.
+	 */
+	public function __construct() {
+		$this->register_yoast_plugins();
+		$this->register_yoast_plugins_status();
+	}
+
+	/**
+	 * Registers all the available Yoast SEO plugins.
+	 */
+	protected function register_yoast_plugins() {
+		$this->plugins = array(
+			'yoast-seo-premium' => array(
+				'url'       => 'https://yoast.com/wordpress/plugins/seo-premium/',
+				'title'     => 'Yoast SEO Premium',
+				/* translators: %1$s expands to Yoast SEO */
+				'description' => sprintf( __( 'The premium version of %1$s with more features & support.', 'wordpress-seo' ), 'Yoast SEO' ),
+				'installed' => false,
+			),
+
+			'video-seo-for-wordpress-seo-by-yoast' => array(
+				'url'       => 'https://yoast.com/wordpress/plugins/video-seo/',
+				'title'     => 'Video SEO',
+				'description' => __( 'Optimize your videos to show them off in search results and get more clicks!', 'wordpress-seo' ),
+				'installed' => false,
+			),
+
+			'yoast-news-seo' => array(
+				'url'       => 'https://yoast.com/wordpress/plugins/news-seo/',
+				'title'     => 'News SEO',
+				'description' => __( 'Are you in Google News? Increase your traffic from Google News by optimizing for it!', 'wordpress-seo' ),
+				'installed' => false,
+			),
+
+			'local-seo-for-yoast-seo' => array(
+				'url'       => 'https://yoast.com/wordpress/plugins/local-seo/',
+				'title'     => 'Local SEO',
+				'description' => __( 'Rank better locally and in Google Maps, without breaking a sweat!', 'wordpress-seo' ),
+				'installed' => false,
+			),
+
+			'yoast-woocommerce-seo' => array(
+				'url'       => 'https://yoast.com/wordpress/plugins/yoast-woocommerce-seo/',
+				'title'     => 'Yoast WooCommerce SEO',
+				/* translators: %1$s expands to Yoast SEO */
+				'description' => sprintf( __( 'Seamlessly integrate WooCommerce with %1$s and get extra features!', 'wordpress-seo' ), 'Yoast SEO' ),
+				'installed' => false,
+			),
+		);
+	}
+
+	/**
+	 * Sets certain plugin properties based on WordPress' status.
+	 */
+	protected function register_yoast_plugins_status() {
+		$installed_plugins = get_plugins();
+
+		foreach ( $installed_plugins as $filename => $plugin ) {
+
+			$plugin_slug = sanitize_title( $plugin['Name'] );
```

Comment:
I believe `sanitize_title` is not something we want to do on each admin request a lot of times. We should just hardcode the slugs.